mne/beamformer/_dics.py

```diff
@@ -446,35 +957,85 @@ def tf_dics(epochs, forward, noise_csds, tmin, tmax, tstep, win_lengths,
     n_ffts : list | None
-        FFT lengths to use for each frequency bin.
+        Length of the FFT for each frequency bin. If ``None`` (the default),
+        the exact number of samples between ``tmin`` and ``tmax`` will be used.
+        Only used in 'multitaper' or 'fourier' mode.
     mt_bandwidths : list of float
         The bandwidths of the multitaper windowing function in Hz. Only used in
         'multitaper' mode. One value should be provided for each frequency bin.
+        Defaults to None.
     mt_adaptive : bool
         Use adaptive weights to combine the tapered spectra into CSD. Only used
-        in 'multitaper' mode.
+        in 'multitaper' mode. Defaults to False.
     mt_low_bias : bool
         Only use tapers with more than 90% spectral concentration within
-        bandwidth. Only used in 'multitaper' mode.
+        bandwidth. Only used in 'multitaper' mode. Defaults to True.
+    cwt_n_cycles: float | list of float | None
+        Number of cycles to use when constructing Morlet wavelets. Fixed number
+        or one per frequency. Defaults to 7.
+        Only used in 'cwt_morlet' mode.
+    decim : int | slice
+        To reduce memory usage, decimation factor during time-frequency
+        decomposition. Defaults to 1 (no decimation).
+        Only used in 'cwt_morlet' mode.
+
+        If `int`, uses tfr[..., ::decim].
+        If `slice`, uses tfr[..., decim].
+    subtract_evoked : bool
+        If True, subtract the averaged evoked response prior to computing the
+        tf source grid. Defaults to False.
     reg : float
-        The regularization for the cross-spectral density.
+        Regularization to use for the DICS beamformer computation.
+        Defaults to 0.05.
     label : Label | None
-        Restricts the solution to a given label.
-    pick_ori : None | 'normal'
-        If 'normal', rather than pooling the orientations by taking the norm,
-        only the radial component is kept.
+        Restricts the solution to a given label. Defaults to None.
+    pick_ori : None | 'normal' | 'max-power'
+        The source orientation to estimate source power for:
+
+            ``None`` :
+                orientations are pooled. (Default)
+            'normal' :
+                filters are computed for the orientation tangential to the
+                cortical surface
+            'max-power' :
+                filters are computer for the orientation that maximizes
+                spectral power.
+
+    beamformer_mode : 'scalar' | 'vector'
+        Whether to compute a scalar or vector beamformer. Defaults to 'scalar'.
+    weight_norm : None | 'unit-noise-gain'
+        How to normalize the beamformer weights. None means no normalization is
+        performed.  If 'unit-noise-gain', the unit-noise gain minimum variance
+        beamformer will be computed (Borgiotti-Kaplan beamformer) [2]_.
+        Defaults to 'unit-noise-gain'.
+    normalize_leadfield : bool
```

Comment:
the option normalize_leadfield is not illustrated in the examples and I am still confused if it's redundant with weight_norm

it's also very related to the depth option which is a float for MNE/dSPM/sLORETA

Reply:
it's also not present in lcmv code so I am wondering if we could remove it...

Reply:
It's close, but not exactly the same as weight normalization. Right now, in order to reproduce our DICS pipeline, we need this parameter. Whether we need it and whether LCMV needs to add the `normalize_leadfield` parameter or DICS needs to remove it, is something we need to find out. Simulations will need to be run, etc. I propose to keep it right in this PR, so we can easily run the needed tests.

Right now, we don't know what the best overall approach is, so we leave it up to the user to make all the decisions for us. Our beamformer expert team (@sarangnemo @britta-wstnr @jjkujala) should get together and figure this out, so our users don't have to.  Hopefully we can resolve it before the next release of MNE to avoid a deprecation cycle.

Reply:
The DICS and LCMV beamformer code should be merged in a next PR. Much of the code path is the same and should be a private function, with two different entry points using either a covariance matrix for LCMV or a CSD for DICS. Before we can do that however, we need to agree on the one and true approach.

Reply:
to replicate @jjkujala local pipeline can you do it with a separate function that would normalize fwd before being passed to dics code?

also we never use leadfield in our parameters / function name. We use the word "forward" or "fwd" to mean leadfield.

Reply:
It's on my TODO list, as such a function would need to be merged with the likes of `compute_depth_prior` and `_prepare_gain_column`. There already seems to be a lot of code  that performs normalization (=depth weighting) of the forward model in some form. It would be great if that could be bundled in a single function somehow.